cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/BudgetsPage.java

```diff
@@ -0,0 +1,324 @@
+        elementMap.put("saveSavingsBucketButton", saveSavingsBucketButton);
+        elementMap.put("confirmDeleteSavingsBucketButton", confirmDeleteSavingsBucketButton);
+        elementMap.put("editSpendingBudgetButton", editSpendingBudgetButton);
+        elementMap.put("totalBudgetAmountInput", totalBudgetAmountInput);
+        elementMap.put("submitSpendingBudgetButton", submitSpendingBudgetButton);
+        elementMap.put("spendingBudgetDiv", spendingBudgetDiv);
+    }
+
+    @Override
+    public List<Component> getChildComponents() {
+        ArrayList<Component> components = new ArrayList<Component>();
+        components.add(navbar);
+        return new ArrayList<Component>(Arrays.asList(navbar));
+    }
+
+    @Override
+    public Component getChildComponent(String name) {
+        if(name.equals("navbar")) return navbar;
+        return null;
+    }
+
+    @Override
+    public List<WebElement> getWebElements() {
+        List<WebElement> webElements = Arrays.asList(summaryComponentDiv, addNewBudgetButton,
+                budgetsTable, savingsBucketTable);
+
+        System.out.println("Web elements: " + webElements);
+
+        return webElements;
+    }
+
+    @Override
+    public WebElement getWebElement(String name) {
+        WebElement element = elementMap.get(name);
+        if (element == null) {
+            throw new IllegalArgumentException("No such element: " + name);
+        }
+        return element;
+    }
+
+    @Override
+    public List<WebElement> getButtons() {
+        return buttonList;
+    }
+
+    @Override
+    public void clickButton(String name) {
+        WebElement button = getWebElement(name);
+
+        if(button == null) throw new IllegalArgumentException("Button '" + name + "' does not exist.");
+        button.click();
+    }
+
+    public void setGroceryActionButtons(String budgetCategory, String budgetedAmount) {
+        List<WebElement> rows = budgetsTable.findElements(By.tagName("tr"));
+
+        for (WebElement row : rows) {
+            List<WebElement> cells = row.findElements(By.tagName("td"));
+            if (cells.size() > 1) {
+                String category = cells.get(0).getText();
+                String amount = cells.get(1).getText();
+                if (budgetCategory.equals(category) && budgetedAmount.equals(amount)) {
+                    this.editBudgetButton = cells.get(5).findElement(By.id("Edit-Budget"));
+                    elementMap.put("editBudgetButton", editBudgetButton);
+                    this.deleteBudgetButton = cells.get(5).findElement(By.id("Delete-Budget"));
+                    elementMap.put("deleteBudgetButton", deleteBudgetButton);
+                    return;
+                }
+            }
+        }
+        throw new IllegalArgumentException(
+                "No " + budgetCategory + " budget found with " + budgetedAmount + " budgeted");
+    }
+
+    public void setEditBudgetModalElements() {
```

Comment:
Low Impact: While performing a similar function to setGroceryActionButtons, this method does not throw an IllegalArgumentException if the budget modal is not found.
Suggestion: Remove throw in setGroceryActionButtons, or add throw to this method.

Edit: Other "set<Something" methods also throw, would recommend adding throw over removing.